### PR TITLE
Migrate PostAggregateAndProofs to MigratingEndpointAdapter

### LINKED ISSUE #5669: Migrate PostAggregateAndProofs to MigratingEndpointAdapter

### PR DESCRIPTION
**Pull request by courtneyeh**
fixes #5669 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostAggregateAndProofs.java [hunk 1]
```diff
@@ -70,18 +109,32 @@ public PostAggregateAndProofs(
...
 
-      handlePostDataResult(ctx, future);
-    } catch (IllegalArgumentException e) {
-      ctx.json(BadRequest.badRequest(jsonProvider, e.getMessage()));
-      ctx.status(SC_BAD_REQUEST);
-    }
+  private static DeserializableTypeDefinition<SignedAggregateAndProof> getRequestBodyType(
+      final SpecConfig specConfig) {
+    return new SignedAggregateAndProof.SignedAggregateAndProofSchema(specConfig)
+        .getJsonTypeDefinition();
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostAggregateAndProofs.java]: should be using `schemaDefinitions.getSignedAggregateAndProofSchema()` basically from schema definition cache i think?

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/schema/ErrorListBadRequest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import tech.pegasys.teku.validator.api.SubmitDataError;
+
+public class ErrorListBadRequest {
+  private static final int CODE = SC_BAD_REQUEST;
+  private final String message;
+  private final List<SubmitDataError> errors;
+
+  public ErrorListBadRequest(final String message, final List<SubmitDataError> errors) {
+    this.message = message;
+    this.errors = errors;
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/schema/ErrorListBadRequest.java:29]: this looks the same as the PostSyncCommittees response, can we use the same classes?

Comment (courtneyeh) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/schema/ErrorListBadRequest.java:29]: Yes, should I do this in a seperate PR?

Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/schema/ErrorListBadRequest.java:29]: can probably just use whats already there without another pr? what would you prefer?

Comment (courtneyeh) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/schema/ErrorListBadRequest.java:29]: Okay, I'll do it in this one

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostAggregateAndProofs.java (hunk 1 above)
Comment (courtneyeh) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostAggregateAndProofs.java]: Updated now. Please check what I have done is what you meant

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/schema/ErrorListBadRequest.java (hunk 2 above)
Comment (courtneyeh) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/schema/ErrorListBadRequest.java:29]: Actually would be cleaner to do in another one, so will do that after this has merged

File: data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostAggregateAndProofsTest.java [hunk 3]
```diff
@@ -13,80 +13,95 @@
...
-  private final DataStructureUtil dataStructureUtil =
-      new DataStructureUtil(TestSpecFactory.createDefault());
-
-  private final Context context = mock(Context.class);
-  private final ValidatorDataProvider provider = mock(ValidatorDataProvider.class);
-  private final JsonProvider jsonProvider = new JsonProvider();
+import tech.pegasys.teku.validator.api.SubmitDataError;
 
-  private PostAggregateAndProofs handler;
+class PostAggregateAndProofsTest extends AbstractMigratedBeaconHandlerTest {
```
Comment (rolfyone) [inline data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostAggregateAndProofsTest.java]: public class?

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostAggregateAndProofs.java [hunk 4]
```diff
@@ -70,18 +110,26 @@ public PostAggregateAndProofs(
...
-          parseRequestBody(ctx.body(), SignedAggregateAndProof[].class);
+  public void handle(@NotNull Context ctx) throws Exception {
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final List<SignedAggregateAndProof> signedAggregateAndProofs = request.getRequestBody();
+    final SafeFuture<List<SubmitDataError>> future =
+        provider.sendAggregateAndProofs(signedAggregateAndProofs);
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostAggregateAndProofs.java:121]: so we're retrieving this list but then using ErrorList still? maybe partial change, not sure, just confused...

Comment (courtneyeh) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostAggregateAndProofs.java:121]: If I wanted to return the `ErrorListBadRequest` type from `sendAggregateAndProofs()`, I would need to add a dependency which I was avoiding, but I can if that's what we want?

Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostAggregateAndProofs.java:121]: we should talk about the dependency... not sure

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**